### ISSUE TITLE
Set-RsDataBase issue - Get-WmiObject error - seems to be tied to SQL 2017/SSRS 2017

### ISSUE THREAD

**Issue by gorkam23** [microsoft/ReportingServicesTools#244]
**Do you want to request a *feature* or report a *bug*?**

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**

**Which versions of Powershell and which OS are affected by this issue? Did this work in previous versions of our scripts?**

**Comment by gorkam23** [comment]
When I tried to move a new SSRS 2017 instance onto a SQL Server 2017 instance, I ran into some odd issues.  

1) When I was on SQL Server 2016/SSRS 2016, I did not have to specify the -ReportServerInstance nor the -ReportServerVersion.   (Github #104 )

I am using PowerShell 5.1 and Windows 10 and SSRS 2017/SQL 2017 and ReportingServicesTools 0.0.5.0

If I don't specify the -ReportServerInstance and -ReportServerVersion, I get the Wmi-NameSpace issue as listed in Github #104 .

Aside from this, I am now getting this odd error.

Confirm
Are you sure you want to perform this action?
Performing the operation "Configure to use localhost as database, using Windows runtime authentication and  setup
authentication" on target "localhost (SQLServer2017) \ SSRS".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): y
Get-WmiObject :
At C:\Program Files\WindowsPowerShell\Modules\ReportingServicesTools\0.0.5.0\Functions\Utilities\New-RsConfigurationSet
tingObject.ps1:100 char:19
+     $wmiObjects = Get-WmiObject @getWmiObjectParameters
+                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Get-WmiObject], COMException
    + FullyQualifiedErrorId : GetWMICOMException,Microsoft.PowerShell.Commands.GetWmiObjectCommand


Here is the script I am trying to run:
Set-RsDatabase -DatabaseServerName localhost -Name ReportServer_ALT -ReportServerInstance SSRS -ReportServerVersion SQLServer2017 -DatabaseCredentialType Windows -DatabaseCredential 'GWML-8KNY4Y2\ReportUser' -VERBOSE

Any ideas?

Thank you!

**Comment by jtarquino** [comment]
For the instance and version is by design , the default for the tools it is 2016 and have not change to avoid breaking anyone that took dependency in the initial version.

The defaults are defined here https://github.com/microsoft/ReportingServicesTools/blob/e5948d5c8a66eaaf86d96ffff0ebe4c2ed0367e6/ReportingServicesTools/Libraries/library.ps1#L24

**Comment by gorkam23** [comment]
No, and I understand that the tools were built around SSRS 2016 and I think enforcing the defaults makes a lot of sense, but I am still getting this odd error.  

Get-WmiObject :
At C:\Program Files\WindowsPowerShell\Modules\ReportingServicesTools\0.0.5.0\Functions\Utilities\New-RsConfigurationSet
tingObject.ps1:100 char:19

$wmiObjects = Get-WmiObject @getWmiObjectParameters
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CategoryInfo : InvalidOperation: (:) [Get-WmiObject], COMException
FullyQualifiedErrorId : GetWMICOMException,Microsoft.PowerShell.Commands.GetWmiObjectCommand

I suspect if I go back to SSRS 2016 that it will work.

**Comment by gorkam23** [comment]
Ok, New-RsConfigurationSettingObject is working at its simplest level with SSRS and SQLServer2017.  It's something else in my script that may be causing the problem.


![new-rsconfigurationsettingobject](https://user-images.githubusercontent.com/32223642/62893119-a2c44e80-bcfe-11e9-9cbc-3b226112c934.png)

**Comment by gorkam23** [comment]
I went ahead and tested on a VM running SQL 2016 and SSRS 2016 and version 0.0.5.0 of the RST.  I was able to use the Set-RsDatabase function without issue on there.

**Comment by gorkam23** [comment]
Any ideas?  Thanks

**Comment by jtarquino** [comment]
It will need a little bit of debugging, probably you will need to add some tracing to write the command used under the cover and use that one independently to figure out what could be causing the issue
